crates/provider/src/ext/anvil.rs

```diff
@@ -0,0 +1,949 @@
+    }
+
+    async fn anvil_set_chain_id(&self, chain_id: u64) -> TransportResult<()> {
+        self.client().request("anvil_setChainId", (chain_id,)).await
+    }
+
+    async fn anvil_set_balance(&self, address: Address, balance: U256) -> TransportResult<()> {
+        self.client().request("anvil_setBalance", (address, balance)).await
+    }
+
+    async fn anvil_set_code(&self, address: Address, code: Bytes) -> TransportResult<()> {
+        self.client().request("anvil_setCode", (address, code)).await
+    }
+
+    async fn anvil_set_nonce(&self, address: Address, nonce: U256) -> TransportResult<()> {
+        self.client().request("anvil_setNonce", (address, nonce)).await
+    }
+
+    async fn anvil_set_storage_at(
+        &self,
+        address: Address,
+        slot: U256,
+        val: B256,
+    ) -> TransportResult<bool> {
+        self.client().request("anvil_setStorageAt", (address, slot, val)).await
+    }
+
+    async fn anvil_set_logging(&self, enable: bool) -> TransportResult<()> {
+        self.client().request("anvil_setLoggingEnabled", (enable,)).await
+    }
+
+    async fn anvil_set_min_gas_price(&self, gas: U256) -> TransportResult<()> {
+        self.client().request("anvil_setMinGasPrice", (gas,)).await
+    }
+
+    async fn anvil_set_next_block_base_fee_per_gas(&self, basefee: U256) -> TransportResult<()> {
+        self.client().request("anvil_setNextBlockBaseFeePerGas", (basefee,)).await
+    }
+
+    async fn anvil_set_coinbase(&self, address: Address) -> TransportResult<()> {
+        self.client().request("anvil_setCoinbase", (address,)).await
+    }
+
+    async fn anvil_dump_state(&self) -> TransportResult<Bytes> {
+        self.client().request("anvil_dumpState", ()).await
+    }
+
+    async fn anvil_load_state(&self, buf: Bytes) -> TransportResult<bool> {
+        self.client().request("anvil_loadState", (buf,)).await
+    }
+
+    async fn anvil_node_info(&self) -> TransportResult<NodeInfo> {
+        self.client().request("anvil_nodeInfo", ()).await
+    }
+
+    async fn anvil_metadata(&self) -> TransportResult<Metadata> {
+        self.client().request("anvil_metadata", ()).await
+    }
+
+    async fn anvil_remove_pool_transactions(&self, address: Address) -> TransportResult<()> {
+        self.client().request("anvil_removePoolTransactions", (address,)).await
+    }
+
+    async fn anvil_snapshot(&self) -> TransportResult<U256> {
+        self.client().request("evm_snapshot", ()).await
+    }
+
+    async fn anvil_revert(&self, id: U256) -> TransportResult<bool> {
+        self.client().request("evm_revert", (id,)).await
+    }
+
+    async fn anvil_increase_time(&self, seconds: U256) -> TransportResult<i64> {
+        self.client().request("evm_increaseTime", (seconds,)).await
+    }
+
+    async fn anvil_set_next_block_timestamp(&self, seconds: u64) -> TransportResult<()> {
+        self.client().request("evm_setNextBlockTimestamp", (seconds,)).await
```

Comment:
some of these are declared as fn anvil but are evm_ ?

Reply:
yeah, these were originally introduced by ganache which is deprecated now, imo we can keep the anvil prefix because these are also aliased in anvil:

https://github.com/foundry-rs/foundry/blob/e764c316c96c8062e9bc52f78f7ee45ea359bc60/crates/anvil/core/src/eth/mod.rs#L559-L559

but these also exist on hardhat iirc

Reply:
Yeah I've used the anvil prefix if it is available. This is the case for most methods but unfortunately not all.